command/task_create.go

```diff
@@ -0,0 +1,181 @@
+package command
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/consul-terraform-sync/api"
+	"github.com/hashicorp/consul-terraform-sync/config"
+	"github.com/mitchellh/go-wordwrap"
+)
+
+const cmdTaskCreateName = "task create"
+
+// TaskCreateCommand handles the `task create` command
+type taskCreateCommand struct {
+	meta
+	autoApprove *bool
+	flags       *flag.FlagSet
+}
+
+func newTaskCreateCommand(m meta) *taskCreateCommand {
+	flags := m.defaultFlagSet(cmdTaskCreateName)
+	a := flags.Bool(FlagAutoApprove, false, "Skip interactive approval of inspect plan")
+	return &taskCreateCommand{
+		meta:        m,
+		autoApprove: a,
+		flags:       flags,
+	}
+}
+
+// Name returns the subcommand
+func (c taskCreateCommand) Name() string {
+	return cmdTaskCreateName
+}
+
+// Help returns the command's usage, list of flags, and examples
+func (c *taskCreateCommand) Help() string {
+	c.meta.setHelpOptions()
+	helpText := fmt.Sprintf(`
+Usage: consul-terraform-sync task create [options] --task-file=<task config>
+
+  Task Create is used to create a new task. It is not to be used for updating a task, it will not create a task if the
+  task name already exists.
+
+Options:
+%s
+
+Example:
+
+  $ consul-terraform-sync task create --task-file="task.hcl"
+  ==> Inspecting changes to resource if creating task 'my_task'...
+
+  // ... inspection details
+
+  ==> Creating the task will perform the actions described above.
+      Do you want to perform these actions for 'my_task'?
+       - This action cannot be undone.
+       - Consul Terraform Sync cannot guarantee that these exact actions will be
+	     performed if monitored services have changed.
+
+      Only 'yes' will be accepted to approve.
+
+  Enter a value: yes
+  
+  // ... output continues
+`, strings.Join(c.meta.helpOptions, "\n"))
+	return strings.TrimSpace(helpText)
+}
+
+// Synopsis is a short one-line synopsis of the command
+func (c *taskCreateCommand) Synopsis() string {
+	return "Creates a new task."
+}
+
+// Run runs the command
+func (c *taskCreateCommand) Run(args []string) int {
+	c.meta.setFlagsUsage(c.flags, args, c.Help())
+	var taskFile string
+	c.flags.StringVar(&taskFile, "task-file", "", "A file containing the hcl definition of a task")
```

Comment:
Our configs support both HCL and JSON, do you know if there is json limitation for the CLI feature?

Reply:
nope you're right, I'll fix this description